src/shims/posix/fs.rs

```diff
@@ -394,9 +424,14 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             && cmd == this.eval_libc_i32("F_FULLFSYNC")?
         {
             let &[_, _] = check_arg_count(args)?;
-            if let Some(FileHandle { file, writable }) = this.machine.file_handler.handles.get(&fd) {
-                let io_result = maybe_sync_file(file, *writable, File::sync_all);
-                this.try_unwrap_io_result(io_result)
+            if let Some(file_descriptor) = this.machine.file_handler.handles.get(&fd) {
+                match file_descriptor.as_file_handle() {
```

Comment:
Same, this should use `?` to produce the right error ("interpreter does not support this operation", instead of an incorrect "handle not found").